### ISSUE TITLE
Require raw values in enum

### ISSUE THREAD

**Issue by jpsim** [realm/SwiftLint#1778]
The opposite of our current rule that triggers a violation if the same string values as enum member names were used, resulting in redundant duplication, would sometimes be useful if you want to allow refactoring of the Swift API without breaking the API contract.

**Comment by marcelofabri** [comment]
This should be a good rule for someone wanting to begin contributing to SwiftLint. `RedundantStringEnumValueRule` can be used as an inspiration (but I'd expect this one to be actually simpler).

I also think this should be opt-in.

**Comment by ornithocoder** [comment]
There's also the `Error` case to thing about.

```swift
enum FooError: Error {
    case expected(number: Int, got: Int)
}
```

**Comment by jpsim** [comment]
Yes, definitely opt-in.
